tools/toolchains/__init__.py

```diff
@@ -188,6 +188,24 @@ def __str__(self):
 }
 
 
+def check_toolchain_path(function):
+    """Check if the path to toolchain is valid. Exit if not.
+    Use this function as a decorator.  Causes a system exit if the path does
+    not exist. Execute the function as normal if the path does exist.
+
+    Positional arguments:
+    function -- the function to decorate
+    """
+    def perform_check(self, *args, **kwargs):
+        if not exists(self.toolchain_path):
+            print('[ERROR] Could not find executable for %s.\n'
+                  'Currently set search path: %s'
+                  % (self.name, self.toolchain_path))
+            sys.exit(-1)
```

Comment:
Did we come to a decision about calling exit within toolchains? I'd still vote no on this and throw an exception instead